tests/client/amo/components/TestOverallRating.js

```diff
@@ -1,31 +1,203 @@
+                   'Some Add-on');
+  });
+
+  it('prompts you to rate the add-on at an exact version', () => {
+    const rootNode = render({ version: { id: 1, version: '2.0.14' } });
     assert.include(rootNode.querySelector('p').textContent,
-                   'some-addon');
+                   '2.0.14');
+  });
+
+  it('creates a rating with add-on and version info', () => {
+    const createRating = sinon.stub();
+    const root = render({
+      createRating,
+      authToken: 'token',
+      version: { id: 321 },
+      addonID: 12345,
+    });
+    selectRating(root, 'input#OverallRating-love-it');
+    assert.equal(createRating.called, true);
+
+    const call = createRating.firstCall.args[0];
+    assert.equal(call.versionID, 321);
+    assert.equal(call.authToken, 'token');
+    assert.equal(call.addonID, 12345);
+  });
+
+  it('lets you submit a "love it" rating', () => {
+    const createRating = sinon.stub();
+    const root = render({ createRating });
+    selectRating(root, 'input#OverallRating-love-it');
+    assert.equal(createRating.called, true);
+    assert.equal(createRating.firstCall.args[0].rating, 5);
+  });
+
+  it('lets you submit a "it is OK" rating', () => {
+    const createRating = sinon.stub();
+    const root = render({ createRating });
+    selectRating(root, 'input#OverallRating-it-is-ok');
+    assert.equal(createRating.called, true);
+    assert.equal(createRating.firstCall.args[0].rating, 3);
+  });
+
+  it('lets you submit a "huh?" rating', () => {
+    const createRating = sinon.stub();
+    const root = render({ createRating });
+    selectRating(root, 'input#OverallRating-huh');
+    assert.equal(createRating.called, true);
+    assert.equal(createRating.firstCall.args[0].rating, 1);
+  });
+
+  it('ignores form changes without a selected rating', () => {
+    const createRating = sinon.stub();
+    const root = render({ createRating });
+    const form = root.querySelector('form');
+    Simulate.change(form, { currentTarget: form });
+    assert.equal(createRating.called, false);
+  });
+
+  describe('mapDispatchToProps', () => {
+    describe('createRating', () => {
+      let mockApi;
+      let actions;
+      let dispatch;
+
+      beforeEach(() => {
+        mockApi = sinon.mock(api);
+        dispatch = sinon.stub();
+        actions = mapDispatchToProps(dispatch);
+      });
+
+      const createRating = (customProps = {}) => actions.createRating({
+        rating: 5,
+        authToken: 'auth-token',
+        addonID: fakeAddon.id,
+        versionID: fakeAddon.current_version.id,
+        ...customProps,
+      });
+
+      it('posts and dispatches the user rating', () => {
+        const params = {
+          rating: 5,
+          authToken: 'auth-token',
+          addonID: 123456,
+          versionID: 321,
+        };
+        const ratingResponse = createRatingResponse({
+          rating: params.rating,
+        });
+
+        mockApi
+          .expects('callApi')
```

Comment:
I think all the other API calls are in `core/api`. This is probably AMO specific so it could go into an `amo/api` file but that might make this test shorter if it just had to call `mockApi.expects('setRating').with({ addonId, rating, versionId })`.